feat: introduce `os_crypt_async` in `safeStorage` (#49054)
electron/electron: docs/api/safe-storage.md

This module adds extra protection to data being stored on disk by using OS-provided cryptography systems. Current
security semantics for each platform are outlined below.

> [!NOTE]

> > We recommend using the asynchronous API (`asyncEncryptString`/`asyncDecryptString`) over the synchronous API.

Reviewer: ```suggestion
> It is recommended to use the asynchronous API (`asyncEncryptString`/`asyncDecryptString`) over the synchronous API.
```
